docs/CODE_GUIDELINES.md

```diff
@@ -0,0 +1,341 @@
+* catch in a try statement
+* while in a do statement
+
+### 10.1. if else
+Put `then`, `return` or `throw` statements on a new line. Keep `else if` statements on one line.
+```cpp
+if (true)
+  return;
+if (true)
+{
+  [...]
+} 
+else if (false)
+{
+  return;
+} 
+else
+  return;
+```
+
+### 10.2. switch case
+```cpp
+switch (cmd)
+{
+  case x:
+  {
+    doSomething();
+    break;
+  }
+  case x:
+  case z:
+    return true;
+  default:
+    doSomething();
+}
+```
+
+**[back to top](#table-of-contents)**
+
+## 11. Naming
+### 11.1. Namespaces
+Namespaces should be in uppercase.
+```cpp
+namespace KODI
+{
+  [...]
+}
+```
+
+### 11.2. Constants
+Use uppercase with underscore spacing where necessary.
+```cpp
+const int MY_CONSTANT = 1;
+```
+
+### 11.3. Enums
+Use CamelCase for the enum name and uppercase for the values.
+```cpp
+enum Dummy
+{
+  VALUE_X,
+  VALUE_Y
+};
+```
+
+### 11.4. Interfaces
+Use CamelCase for interface names and they should be prefixed with an uppercase I. Filename should match the interface name without the prefixed I, e.g. ILogger.h
+```cpp
+class ILogger
+{
+  void Log(...) = 0;
+}
+```
+
+### 11.5. Classes
+Use CamelCase for class names and they should be prefixed with an uppercase C. Filename should match the class name without the prefixed C, e.g. Logger.cpp
+```cpp
+class CLogger : public ILogger
+{
+  void Log(...)
+}
+```
+
+### 11.6. Methods
+Use CamelCase for method names and first letter should always be uppercase, even if the methods are private or protected.
+```cpp
+void MyDummyClass::DoSomething();
+```
+
+### 11.7. Variables
+Use CamelCase for variables. Type prefixing is discouraged.
+
+#### Global Variables
+Prefix global variables with *g_*
+```cpp
+int g_globalVariableA;
+```
+**WARNING:** Avoid globals use. It increases the chances of submitted code to be rejected.
+
+#### Member Variables
+Prefix member variables with *m_*
+```cpp
+int m_variableA;
+```
+
+**[back to top](#table-of-contents)**
+
+## 12. Conventions
+### 12.1. Casts
+New code should use C++ style casts and not older C style casts. When modifying existing code the developer can choose to update it to C++ style casts or leave as is. Remember that whenever a `dynamic_cast` is used on a pointer object the result can be a `nullptr` and needs to be checked accordingly.
+
+### 12.2. NULL vs nullptr
+Prefer the use of `nullptr` instead of `NULL`. `nullptr` is a typesafe version and as such can't be implicitly converted to `int` or anything else.
+
+### 12.3. auto
+
+Feel free to use `auto` wherever it improves readability. Good places are iterators or when dealing with containers.
+```cpp
+std::map<std::string, std::vector<int>>::iterator i = var.begin();
+```
+vs
+```cpp
+auto i = var.being();
+```
+### 12.4. for loops
+Use range-based for loops wherever it makes sense. If iterators are used see above about using `auto`.
+```cpp
+for (auto& : var)
+{
+  [...]
+}
+```
+Use `const auto&` if there's no reason to modify the value.
```

Comment:
the multiline example should use `const auto&` and the note should mention to remove the const if value has to be modified

Reply:
Fixed